tests/e2e/s3/utils.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+
+import subprocess
+import json
+import os
+import sys
+import boto3
+import time
+
+
+def update_agent_biling_clients(billing):
```

Comment:
From reading `agent_remove_s3`, seems like this function is mis-named - IIUC it currently updates all of the billing config, but it's called `update_agent_billing_clients`. IMO just updating the billing clients will allow the calling code to be simpler and more readable

Reply:
Why is updating the other fields required? Is it just that otherwise the test will take too long?

(if so: maybe it'd be preferable to keep the same settings we use in staging/prod, even if the test takes longer — wdyt?)

Reply:
>  Is it just that otherwise the test will take too long?

Yes, that was my reasoning.

> even if the test takes longer — wdyt?

I think, it is important for developer productivity to have tests which run as fast as possible. I don't see a case where keeping the default period values helps us catch something we wouldn't have caught otherwise.

Reply:
The idea behind keeping the default period would be that we can test the values are actually what we expect. There's rounding that happens, so in this case the correctness is impacted by what period is used.

At the same time: if we're not testing it the actual values, we don't get any benefit from using the default period. Happy to leave that for a follow-up.